src/user_input/gamepad.rs

```diff
@@ -0,0 +1,1150 @@
+//! Gamepad inputs
+
+use bevy::prelude::{
+    Gamepad, GamepadAxis, GamepadAxisType, GamepadButton, GamepadButtonType, Reflect, Vec2,
+};
+use leafwing_input_manager_macros::serde_typetag;
+use serde::{Deserialize, Serialize};
+
+use crate as leafwing_input_manager;
+use crate::axislike::{AxisDirection, DualAxisData};
+use crate::clashing_inputs::BasicInputs;
+use crate::input_processing::{
+    AxisProcessor, DualAxisProcessor, WithAxisProcessingPipelineExt,
+    WithDualAxisProcessingPipelineExt,
+};
+use crate::input_streams::InputStreams;
+use crate::raw_inputs::RawInputs;
+use crate::user_input::{InputControlKind, UserInput};
+
+/// Retrieves the current value of the specified `axis`.
+#[must_use]
+#[inline]
+fn read_axis_value(input_streams: &InputStreams, axis: GamepadAxisType) -> f32 {
+    let gamepad_value_self = |gamepad: Gamepad| -> Option<f32> {
+        let axis = GamepadAxis::new(gamepad, axis);
+        input_streams.gamepad_axes.get(axis)
+    };
+
+    if let Some(gamepad) = input_streams.associated_gamepad {
+        gamepad_value_self(gamepad).unwrap_or_default()
+    } else {
+        input_streams
+            .gamepads
+            .iter()
+            .filter_map(gamepad_value_self)
+            .find(|value| *value != 0.0)
+            .unwrap_or_default()
+    }
+}
+
+/// Provides button-like behavior for a specific direction on a [`GamepadAxisType`].
+///
+/// # Behaviors
+///
+/// - Gamepad Selection: By default, reads from **any connected gamepad**.
+///     Use the [`InputMap::set_gamepad`] for specific ones.
+/// - Activation: Only if the axis is currently held in the chosen direction.
+/// - Single-Axis Value:
+///   - `1.0`: The input is currently active.
+///   - `0.0`: The input is inactive.
+///
+/// [`InputMap::set_gamepad`]: crate::input_map::InputMap::set_gamepad
+///
+/// ```rust,ignore
```

Comment:
Doc tests for this module are ignored, due to the excessive code required to add usable handles for each example.
I feel that they should be focusing on describing input behavior